### ISSUE TITLE
Where are the Sync Folder Settings?

### ISSUE THREAD

**Issue by WegnerDan** [cp2004/OctoPrint-OneDriveFileSync#6]
I've just installed 1.0.2 and can't find the Sync Folder Settings.

I also use your Plugin "OneDrive Backup", where I've already added my OneDrive account.

![image](https://user-images.githubusercontent.com/6908247/211175216-1670c986-1b5d-41d1-ba35-99fb41450cd2.png)

![image](https://user-images.githubusercontent.com/6908247/211175299-5504c62d-8322-4f73-a6e7-829666d8ce10.png)

**Comment by cp2004** [comment]
As you can imagine, I did copy some of the UI elements from the other plugin. It worked well in development mode, but when built for production (as the releases are automatically) it broke due to a copy-paste error. Lovely. Looks like I need to pay more attention - sorry about that!

**Comment by cp2004** [comment]
Fixed with 1.0.3, btw, which ~~is currently~~ has just finished building.

**Comment by WegnerDan** [comment]
Thanks man. 

I think it's not quite working yet.

Also, take your time, you don't owe us anything!

![firefox_OI8lqIqhtl](https://user-images.githubusercontent.com/6908247/211176658-fac99399-554d-4f20-ac46-06119c2bacc8.gif)

**Comment by ryancasler** [comment]
It worked for me.  I just installed and set up sync and it synced my files.  But then I went in to try and change the folder and now it shows the same error.   I tried forgetting my account as well and that didn't work either.   I turned on logging for the plug-in but it doesn't appear that anything was logged.  
Also, the "Copy Code" button doesn't actually copy the code generated.

**Comment by RussPitcher** [comment]
Just found this plugin and installed it.  Then it immediately upgraded to 1.0.3.  I am now having the same issue in that clicking the 'Set sync folder' buttons don't do anything, they just cause a very quick refresh but the sync folder doesn't get sent.

**Comment by TimDowker** [comment]
Same error here - tried in both Chrome and Safari - no luck.

**Comment by cp2004** [comment]
I've spent some time working on this and have been able to reproduce it now - diving in to find out why it is not sticking, as everything *claims* to be working successfully. It's interestingly only reproduceable on one of my installs (& none of the several development ones), but now that it has done I will be able to find out why.

**Comment by cp2004** [comment]
It does seem to set the OneDrive folder ID in the plugin however and I was able to sync the folder. It just never sets/shows the name (cosmetic only - albeit fairly critical).

**Comment by cp2004** [comment]
I believe I've fixed this issue - please could you test the development version. You should be able to get it either through switching to 'Release Candidate' in the software update settings for this plugin, or by using the below URL in the plugin manager > get more > ...from URL area:

```
https://github.com/cp2004/OctoPrint-OneDriveFileSync/releases/download/1.0.4.dev1/release.zip
```

Thanks for your patience with this - we will get this fully working for everyone eventually!

**Comment by ryancasler** [comment]
Worked for me!  Thanks so much!!!

**Comment by RussPitcher** [comment]
Thanks for that.  That change has allowed me to set the sync folder, but the sync still isn't doing anything.  I've currently set it to sync from Octoprint to Onedrive to populate onedrive with all my current files. I was then intending to reverse the sync so I could manage things easily from the PC.  I've tried unchecking the 'disable sync while printing' (I'm not printing) and alwo played with the automatic syncing checkbox and interval times.  I've also clicked the manual sync button, but still nothing has been copied to the onedrive folder.

Any ideas?
